addons/account/models/account_move.py

```diff
@@ -1293,8 +1293,8 @@ def _compute_tax_lock_date_message(self):
                 accounting_date = move._get_accounting_date(invoice_date, affects_tax_report)
                 lock_date, lock_type = lock_dates[-1]
                 tax_lock_date_message = _(
-                    "The accounting date being set prior to the %(lock_type)s lock date %(lock_date)s,"
-                    " it will be changed to %(accounting_date)s upon posting.",
+                    "The invoice date is being set prior to the %(lock_type)s lock date %(lock_date)s. "
+                    "The invoice will be accounted on %(accounting_date)s upon posting.",
```

Comment:
Mmmh, yes but that message can also happen on journal entries...
And it is the accounting date that matters, not the invoice date